stacks/go/2.0.0/devfile.yaml

```diff
@@ -0,0 +1,81 @@
+schemaVersion: 2.2.0
+metadata:
+  description: 'Go is an open source programming language that makes it easy to build simple, reliable, and efficient software.'
+  displayName: Go Runtime
+  icon: https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/golang.svg
+  name: go
+  projectType: Go
+  provider: Red Hat
+  language: Go
+  tags:
+    - Go
+  version: 2.0.0
+starterProjects:
+  - name: go-starter
+    description: A Go project with a simple HTTP server
+    git:
+      checkoutFrom:
+        revision: main
+      remotes:
+        origin: https://github.com/devfile-samples/devfile-stack-go.git
+components:
+  - name: build
+    image:
+      imageName: go-image:latest
+      dockerfile:
+        uri: docker/Dockerfile
+        buildContext: .
+        rootRequired: false
+  - name: deploy
+    attributes:
+      deployment/replicas: 1
```

Comment:
What are those attributes? Is this specific to only one tool?

Reply:
Those attributes are understood by App Studio and ODC as they were trying to support outerloop based on the devfile 2.1+. With the devfile 2.2 stacks, they are moving toward formally supporting devfile 2.2 so that information should be able to extract from the kubernetes.yaml without the attributes. Moving forward, App Studio will still need a way to be able to do override on the Kubernetes component in a way similar to https://github.com/devfile/api/issues/920. However, I have created https://github.com/devfile/api/issues/988 to track that.

@mike-hoang For these devfile 2.2 devfile, if you can confirm the same settings (and values) are specified on the corresponding kubernetes.yaml in the Kubernetes component, you can remove those attributes.